### PR TITLE
core: update a few sections 

### PR DESCRIPTION
**Pull request by cmwaters**
This PR tidies up a few parts in the `spec/core` section.

- Update the state file, adding information on validator set and consensus params update as well as removing redundant information covered in data structures
- fix the table of contents in `spec/core`
- minor touch ups

### CODE REVIEW

File: spec/core/data_structures.md [hunk 1]
```diff
@@ -26,10 +26,7 @@ The Tendermint blockchains consists of a short list of data types:
 - [`Validator`](#validator)
 - [`ValidatorSet`](#validatorset)
 - [`Address`](#address)
-- [`ConsensusParams](#consensusparams)
-- [`EvidenceParams`](#evidenceparams)
-- [`ValidatorParams`](#validatorparams)
-- [`VersionParams`](#versionparams)
```
Comment (marbar3778) [inline spec/core/data_structures.md:32]: why remove these?

Comment (cmwaters) [reply spec/core/data_structures.md:32]: I considered them under the umbrella of `ConsensusParams`. We can add them back again but in that case we are missing `BlockParams`

Comment (marbar3778) [reply spec/core/data_structures.md:32]: Yes, block params is missing. I would be in favor of leaving them, technically header, evidence data is under the umbrella of block, but we still have them.

Comment (cmwaters) [reply spec/core/data_structures.md:32]: Sure, I will update them

File: spec/core/data_structures.md [hunk 2]
```diff
@@ -446,21 +452,21 @@ func SumTruncated(bz []byte) []byte {
...
 
 | Name          | Type            | Description                                                           | Field Number |
 |---------------|-----------------|-----------------------------------------------------------------------|--------------|
 | pub_key_types | repeated string | List of accepted public key types. Uses same naming as `PubKey.Type`. | 1            |
 
-### VersionParams
+## VersionParams
 
 | Name        | Type   | Description                   | Field Number |
 |-------------|--------|-------------------------------|--------------|
```
Comment (cmwaters) [inline spec/core/data_structures.md:472]: What's the `Proof` below here for. Is it part of `PartSet`?

Comment (cmwaters) [reply spec/core/data_structures.md:472]: Sorry I can't comment the part that I'm referring to directly

Comment (marbar3778) [reply spec/core/data_structures.md:472]: Yes, its part of the part of PartSet (🤔).

**Review by marbar3778 — APPROVED** [review]

File: spec/core/data_structures.md (hunk 2 above)
Comment (cmwaters) [reply spec/core/data_structures.md:472]: Is `PartSet` something that should be in the spec? Do we have a section on block propagation?

**Pull request merged**